### ISSUE TITLE
Type 'object' could not be resolved in Cpp files (IEP-464)

### ISSUE THREAD

**Issue by carlessole** [espressif/idf-eclipse-plugin#290]
Hello,
For my purposes I need to write a program using Cpp (not plain C).

So I started using the [blink ](https://github.com/espressif/esp-idf/tree/v4.2.1/examples/get-started/blink) example.

Everything was OK: project compiles and I was able to run & debug project without problems. The indexer was working OK after building the project (sometimes before build the project some includes were not found and some types could not be resolved).

As I want to work in Cpp I have done the following:

**main.c:**
```
/* Blink Example

   This example code is in the Public Domain (or CC0 licensed, at your option.)

   Unless required by applicable law or agreed to in writing, this
   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
   CONDITIONS OF ANY KIND, either express or implied.
*/
#include <custom_main.h>
#include <stdio.h>
#include "freertos/FreeRTOS.h"
#include "freertos/task.h"
#include "driver/gpio.h"
#include "sdkconfig.h"


void app_main(void)
{
	custom_main();

	//This code should never be reached
    while (1)
    {
    	vTaskDelay(1000 / portTICK_PERIOD_MS);
    }
}
```

Then I create my custom_main:

**custom_main.h:**
```
#ifndef MAIN_CUSTOM_MAIN_H_
#define MAIN_CUSTOM_MAIN_H_

#ifdef __cplusplus
extern "C"
{
#endif

void custom_main(void);

#ifdef __cplusplus
}
#endif

#endif /* MAIN_CUSTOM_MAIN_H_ */
```

**custom_main.cpp:**
```
extern "C"
{
#include <stdio.h>
#include "freertos/FreeRTOS.h"
#include "freertos/task.h"
#include "driver/gpio.h"
#include "sdkconfig.h"

#include <custom_main.h>
}

#define BLINK_GPIO GPIO_NUM_2

void custom_main(void)
{
    /* Configure the IOMUX register for pad BLINK_GPIO (some pads are
       muxed to GPIO on reset already, but some default to other
       functions and need to be switched to GPIO. Consult the
       Technical Reference for a list of pads and their default
       functions.)
    */
    gpio_pad_select_gpio(BLINK_GPIO);
    /* Set the GPIO as a push/pull output */
    gpio_set_direction(BLINK_GPIO, GPIO_MODE_OUTPUT);
    while (1)
    {
        /* Blink off (output low) */
        printf("Turning off the LED\n");
        gpio_set_level(BLINK_GPIO, 0);
        vTaskDelay(1000 / portTICK_PERIOD_MS);
        /* Blink on (output high) */
        printf("Turning on the LED\n");
        gpio_set_level(BLINK_GPIO, 1);
        vTaskDelay(1000 / portTICK_PERIOD_MS);
    }
}
```
And finally, **CMakeList.txt:**
```
idf_component_register(SRCS "main.c" "custom_main.cpp"
                    INCLUDE_DIRS ".")
```

After these changes project is still compiling and I'm able to run it and also debug. But now indexer is complaining about "Type 'object' could not be resolved " on the cpp file:

![imagen](https://user-images.githubusercontent.com/46353439/121408161-61d00c00-c960-11eb-8ca1-a58717028227.png)

And each time I run the program this "error" is reported although if I press "Proceed" everything is OK:

![imagen](https://user-images.githubusercontent.com/46353439/121407466-a4451900-c95f-11eb-9614-0c1a855b0586.png)

The problem is that programing without the help of the indexer is a pain and also seeing always all these errors.

Is it a known issue?

Is there a solution?

Thanks!

**Comment by ivorss** [comment]
Yes, this is very annoying. Need the "Path and Symbols" section under "C/C++General", so I can tell Eclipse where to find the include files.

**Comment by kolipakakondal** [comment]
Hello All, Indexer problems are got fixed with IEP 2.4.0 release. Please get the latest changes using the update site to verify. Thank you.

**Comment by carlessole** [comment]
Ho @kolipakakondal ,
Thanks.
The indexer problems have been solved but now I'm not able to debug my code:

![imatge](https://user-images.githubusercontent.com/46353439/154123385-4e942b37-432f-4a7b-943e-f23424e3377f.png)

Anyone else has faced against that issue with latest update?
